worker/caasoperator/remotestate/watcher.go

```diff
@@ -81,6 +82,12 @@ func (w *RemoteStateWatcher) Snapshot() Snapshot {
 }
 
 func (w *RemoteStateWatcher) loop() (err error) {
+	defer func() {
+		if errors.IsNotFound(err) {
+			err = jworker.ErrTerminateAgent
```

Comment:
Only IAAS models can have the agent terminated. With k8s models, all the uniters run as workers and we need to keep the agent running. For k8s uniter workers, we simple exit the worker.

Reply:
ic, probably we can return `jworker.ErrTerminateAgent` for application not found?

Reply:
yeah, changed to exit peacefully if it got a not found error.